torch/ao/quantization/fx/_model_report/detector.py

```diff
@@ -0,0 +1,408 @@
+            Set[str] of observer fqns denoting where to insert observers
+            ObserverBase (or subclass): the class (not an instance) of the observer to insert
+        """
+        pass
+
+    @abstractmethod
+    def get_detector_name(self):
+        r""" Returns the name of the current detector """
+        pass
+
+    @abstractmethod
+    def generate_detector_report(self, model):
+        r"""
+        Args
+            model (nn.Module or subclass): model to find observer insertion points
+
+        Returns a Tuple of two elements:
+            Str: string report of the suggested improvements
+            Dict: contains useful data collected by the observer pertinent to this report
+        """
+        pass
+
+class PerChannelDetector(DetectorBase):
+    r""" This class is used to detect if any Linear or Conv layers in a model utilize per_channel quantization.
+        Only Linear and Conv layers can use per_channel as of now so only these two are currently checked.
+
+        per_channel quantization can lead to major benefits in the form of accuracy.
+        Therefore, if the backend used by the user supports it, it is recommended to use
+
+        Args:
+            backend (str): the backend the user wishes to use in production
+                If the user wishes to use the current backend, they should set backend = torch.backends.quantized.engine
+    """
+
+    # Default map for representing supported per channel quantization modules for different backends
+    DEFAULT_BACKEND_PER_CHANNEL_SUPPORTED_MODULES: Dict[str, Set[Any]] = {
+        "fbgemm": set([nn.Linear, nn.Conv1d, nn.Conv2d, nn.Conv3d, nnqat.Linear, nnqat.Conv1d, nnqat.Conv2d, nnqat.Conv3d]),
+        "qnnpack": set([nn.Linear, nn.Conv1d, nn.Conv2d, nn.Conv3d, nnqat.Linear, nnqat.Conv1d, nnqat.Conv2d, nnqat.Conv3d]),
+        "onednn": set([nn.Linear, nn.Conv1d, nn.Conv2d, nn.Conv3d, nnqat.Linear, nnqat.Conv1d, nnqat.Conv2d, nnqat.Conv3d]),
+    }
+
+    def __init__(self, backend):
+        super().__init__()
+
+        # store the backend information
+        self.backend_chosen = backend
+        self.supported_modules = set([])
+        if self.backend_chosen in self.DEFAULT_BACKEND_PER_CHANNEL_SUPPORTED_MODULES:
+            self.supported_modules = self.DEFAULT_BACKEND_PER_CHANNEL_SUPPORTED_MODULES[self.backend_chosen]
+        else:
+            raise ValueError("Not configured to work with {}. Try a different default backend".format(self.backend_chosen))
+
+    def get_detector_name(self) -> str:
+        r""" returns the string name of this detector"""
+        return "per_channel_detector"
+
+    def determine_observer_insert_points(self, model):
+        r"""
+        There is no observers inserted for the PerChannelDetector
+        """
+        raise NotImplementedError("No observers are inserted in the PerChannelDetector.")
+
+
+    def __detect_per_channel_helper(self, model: nn.Module, per_channel_info: Dict):
```

Comment:
nit: I think we usually just use 1 underscore for private methods

Reply:
Just checked locally and it has to be two. when i started the methods with only 1 underscore, they were still accessible publically when I tried to call them after creating an instance.

Reply:
looked up docs and I stand corrected, guess it should be one, i will fix